derive/src/field/mod.rs

```diff
@@ -0,0 +1,601 @@
+    let mut inv64 = 1u64;
+    for _ in 0..63 {
+        inv64 = inv64.wrapping_mul(inv64);
+        inv64 = inv64.wrapping_mul(modulus_limbs[0]);
+    }
+    inv64 = inv64.wrapping_neg();
+
+    let mut by_inverter_constant: usize = 2;
+    loop {
+        let t = BigUint::from(1u64) << (62 * by_inverter_constant - 64);
+        if t > modulus {
+            break;
+        }
+        by_inverter_constant += 1;
+    }
+
+    let mut jacobi_constant: usize = 1;
+    loop {
+        let t = BigUint::from(1u64) << (64 * jacobi_constant - 31);
+        if t > modulus {
+            break;
+        }
+        jacobi_constant += 1;
+    }
+
+    let mut s: u32 = 0;
+    let mut t = &modulus - BigUint::one();
+    while t.is_even() {
+        t >>= 1;
+        s += 1;
+    }
+
+    let two_inv = mod_inv(&BigUint::from(2usize), &modulus);
+
+    let sqrt_impl = {
+        if &modulus % 16u64 == BigUint::from(1u64) {
+            let tm1o2 = ((&t - 1usize) * &two_inv) % &modulus;
+            let tm1o2 = big_to_token(&tm1o2, num_limbs);
+            quote! {
+                fn sqrt(&self) -> subtle::CtOption<Self> {
+                    ff::helpers::sqrt_tonelli_shanks(self, #tm1o2)
+                }
+            }
+        } else if &modulus % 4u64 == BigUint::from(3u64) {
+            let exp = (&modulus + 1usize) >> 2;
+            let exp = big_to_token(&exp, num_limbs);
+            quote! {
+                fn sqrt(&self) -> subtle::CtOption<Self> {
+                    use subtle::ConstantTimeEq;
+                    let t = self.pow(#exp);
+                    subtle::CtOption::new(t, t.square().ct_eq(self))
+                }
+            }
+        } else {
+            panic!("unsupported modulus")
+        }
+    };
+
+    let root_of_unity = mul_gen.modpow(&t, &modulus);
+    let root_of_unity_inv = mod_inv(&root_of_unity, &modulus);
+    let delta = mul_gen.modpow(&(BigUint::one() << s), &modulus);
+
+    let root_of_unity = to_token(&mont(&root_of_unity));
+    let root_of_unity_inv = to_token(&mont(&root_of_unity_inv));
+    let two_inv = to_token(&mont(&two_inv));
+    let mul_gen = to_token(&mont(&mul_gen));
+    let delta = to_token(&mont(&delta));
+    let zeta = to_token(&mont(&zeta));
+
+    let endian = match endian.as_str() {
+        "little" => {
+            quote! { LE }
+        }
+        "big" => {
+            quote! { BE }
+        }
+        _ => {
+            unreachable!()
+        }
+    };
+
+    let impl_field = quote! {
+        #[derive(Clone, Copy, PartialEq, Eq, Hash, Default)]
+        pub struct #field(pub(crate) [u64; #num_limbs]);
+
+        impl core::fmt::Debug for #field {
+            fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+                use ff::PrimeField;
+                let tmp = self.to_repr();
+                write!(f, "0x")?;
+                for &b in tmp.as_ref().iter().rev() {
+                    write!(f, "{:02x}", b)?;
+                }
```

Comment:
Right. At the following BLS PR `to_repr` will always return little endian bytes but to_bytes will give the configured endianness result. I made that decision because to_repr is also used in some arithmetic operations related to bits of this element. I'd rather keep this here not to revert it again

Reply:
Got it, sounds good to keep it and update in following PR.